tests/Controller/ModelCollectorTest.php

```diff
@@ -206,6 +214,328 @@ public function testCollectSiblingsOf()
+            $this->createModel('child', 1, ['pid' => 3]),
+            $collection
+        ];
+    }
+
+    /**
+     * Tests the searchParentOfIn method without recursion.
+     *
+     * @param ModelInterface|null             $expected   The expected parent.
+     * @param ModelInterface                  $model      The given instance of the model.
+     * @param CollectionInterface             $candidates The given candidates of the parent for the model.
+     *
+     * @return void
+     *
+     * @dataProvider provideForTestSearchParentOfInWithoutRecursion
+     */
+    public function testSearchParentOfInWithoutRecursion(
+        ?ModelInterface $expected,
+        ModelInterface $model,
+        CollectionInterface $candidates
+    ): void {
+        $definition = $this->mockDefinitionContainer();
+
+        $basicDefinition = $this->mockBasicDefinition();
+        $basicDefinition->method('getDataProvider')->willReturn('child');
+        $basicDefinition->method('getParentDataProvider')->willReturn('parent');
+        $basicDefinition->method('getMode')->willReturn(BasicDefinitionInterface::MODE_PARENTEDLIST);
+        $definition->method('getBasicDefinition')->willReturn($basicDefinition);
+
+        $relationships = $this->mockRelationshipDefinition();
+        $conditions    = [$this->createParentChildCondition('parent', 'child')];
+        $relationships->method('getChildConditions')->willReturn($conditions);
+        $definition->method('getModelRelationshipDefinition')->willReturn($relationships);
+
+        $environment = $this->getMockForAbstractClass(EnvironmentInterface::class);
+        $environment->method('getDataDefinition')->willReturn($definition);
+
+        $config = $this->getMockForAbstractClass(ConfigInterface::class);
+        $config
+            ->method('setFilter')
+            ->willReturn($config);
+
+        $parentProvider = $this->getMockForAbstractClass(DataProviderInterface::class);
+        $parentProvider->method('getEmptyConfig')->willReturn($config);
+        $parentProvider->method('fetchAll')->with($config)->willReturn(new DefaultCollection());
+
+        $provider = $this->getMockForAbstractClass(DataProviderInterface::class);
+        $provider->method('getEmptyConfig')->willReturn($config);
+        $provider->method('fetchAll')->with($config)->willReturn(new DefaultCollection());
+
+        $environment->method('getDataProvider')->willReturnCallback(
+            function (string $name) use ($provider, $parentProvider) {
+                switch ($name) {
+                    case 'child':
+                        return $provider;
+                    case 'parent':
+                        return $parentProvider;
+
+                    default:
+                        throw new \RuntimeException();
+                }
+            }
+        );
+
+        $collector = new ModelCollector($environment);
+
+        $this->assertSame($expected, $collector->searchParentOfIn($model, $candidates));
```

Comment:
Please call static method via `self::assertSame()`

Reply:
Fixed in 7826ab36bea4d62ae0edbd3b8c1e06bdbe44bc44